services/business/checkers/route_rule_checker.go

```diff
@@ -0,0 +1,58 @@
+package checkers
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/business/checkers/route_rules"
+	"github.com/kiali/kiali/services/models"
+)
+
+const objectType = "routerule"
+
+type Checker interface {
+	Check() ([]*models.IstioCheck, bool)
+}
+
+type RouteRuleChecker struct {
+	RouteRules []kubernetes.IstioObject
+}
+
+func (in RouteRuleChecker) Check() *models.IstioValidations {
+	objectValidations := models.IstioValidations{}
+
+	objectValidations = objectValidations.MergeValidations(in.runIndividualChecks())
+	objectValidations = objectValidations.MergeValidations(in.runGroupChecks())
+
+	return &objectValidations
+}
+
+func (in RouteRuleChecker) runIndividualChecks() *models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	for _, routeRule := range in.RouteRules {
+		valid := true
+		ruleName := routeRule.GetObjectMeta().Name
+		istioValidation := &models.IstioValidation{Name: ruleName, ObjectType: objectType}
+
+		for _, checker := range in.enabledCheckersFor(routeRule) {
+			checks, validChecker := checker.Check()
+			istioValidation.Checks = append(istioValidation.Checks, checks...)
+			valid = valid && validChecker
+		}
+
+		istioValidation.Valid = valid
+		validations[ruleName] = istioValidation
+	}
+
+	return &validations
+}
+
+func (in *RouteRuleChecker) runGroupChecks() *models.IstioValidations {
```

Comment:
What is the purpose of this method ?

Reply:
I just divided the checks in two groups:
1. Individual object checks: checking the integrity of that specific object. Not considering how the other co-living  objects modifies its integrity (e.g. validate precedence values - positive number)
2. Object group checks: checking the integrity among a collection of istio objects. Considering the behavior of the whole group. (e.g. route rule with precedence 0 has 100% of the traffic weight, so route rule with precedence 1 has no effect)

Reply:
Ok, so, this runGroupChecks is part of the internal architecture to validate inter-validation between objects. But it is not yet implemented.
The example you describe:

> (e.g. route rule with precedence 0 has 100% of the traffic weight, so route rule with precedence 1 has no effect)

It is not yet in place, isn't it ?

Reply:
> Ok, so, this runGroupChecks is part of the internal architecture to validate inter-validation between objects. But it is not yet implemented.

That is correct.

> It is not yet in place, isn't it ?

No, it isn't. I just preferred to start with it. Not to mix too many things.